packages/block-library/src/gallery/deprecated.js

```diff
@@ -15,6 +15,139 @@ import { RichText } from '@wordpress/block-editor';
 import { defaultColumnsNumber } from './shared';
 
 const deprecated = [
+	{
+		attributes: {
```

Comment:
Is there any opportunity for code reuse in any of these properties or functions? Merging attributes using `Object.assign` / spread? Reference a common `save` function?

This is a lot of code to coerce an array to a typed array.

Reply:
I considerer shared code with the current version of save, but then I saw this comment on the [block deprecations documentation](https://developer.wordpress.org/block-editor/developers/block-api/block-deprecation/): 

> It’s important to note that attributes, supports, and save are not automatically inherited from the current version, since they can impact parsing and serialization of a block, so they must be defined on the deprecated object in order to be processed during a migration.

Is this still a valid concern? Or I can share code without problems with the current version of save?

Reply:
The problem is that if you inherit from the current attributes, the deprecations may work now but not later if more attributes are added to the block. If you add an optional `thingy` attribute to the current block, and your deprecations inherit the entire attributes object from the current implementation, then all the deprecations will stop working. If you change an existing attribute, you have to remember to modify every single deprecation to stop inheriting from the current implementation.

Hardcoding the deprecation definitions, although verbose, is a more robust solution. I don't really see it as much of problem since the deprecations are now usually put in a separate file from the rest of the block data.

Reply:
Yeah, I prefer the longer safer option. Though it's generally ok if there are tests in place.

Which is something I think should be considered for this PR, ideally a `core__gallery__deprecated-4.html` fixture would be added in this folder:
https://github.com/WordPress/gutenberg/tree/master/packages/e2e-tests/fixtures/blocks